### ISSUE TITLE
Keras implementation should show test accuracy while learning

### ISSUE THREAD

**Issue by RobertCram** [RobertCram/machine-learning#4]

**Comment by RobertCram** [comment]
We should probably discuss this. Showing the test error while learning could influence the choice of hyperparameters, thereby giving overly optimistic results on the final test score. Perhaps we can use some sort of validation scheme.
